src/partest/scala/tools/partest/nest/DirectCompiler.scala

```diff
@@ -134,7 +135,22 @@ class DirectCompiler(val runner: Runner) {
         result
       }
 
-    try     { execCompile() }
+    try     {
+        if (suiteRunner.config.optCompilerPath.isEmpty)
+          execCompile()
+        else {
```

Comment:
could you put the code in a separate method, next to `execCompile`?

Reply:
Refactored into a method, thanks.  Also added a comment about the flag not being used during normal Scala testing, plus a slight rewording of the --help text to make that clearer.  You’re definitely right about the slowness of running it this way (I generally run it overnight), but it’s been worth it—the tests have taught us a lot about handling non-blockchain Scala code.